sdk/iot/Azure.Iot.Hub.Service/src/IfMatchPrecondition.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Iot.Hub.Service
+{
+    /// <summary>
+    /// The condition on which an operation will be executed against a service resource. Each value maps to a possible use of the If-Match header. The If-Match header is described in RFC7232.
+    /// </summary>
+    public enum IfMatchPrecondition
+    {
+        /// <summary>
+        /// Perform this operation regardless of if the provided resource matches the service's representation
+        /// of the object. This will cause the HTTP request to be sent with no ifMatch header. The service will never respond with a 412 error code with this setting.
+        /// </summary>
+        Unconditional,
+
+        /// <summary>
+        /// Perform this operation as long as the provided resource exists in the service. This will cause the HTTP request to be sent with an ifMatch header with value "*". For create or update
+        /// operations, if the resource does not exist, then the service will not execute the operation and will respond to the request with a 412 error code.
```

Comment:
Should we also specify what happens on "delete" with "*" (when the resource doesn't exist).